src/android/Capture.java

```diff
@@ -399,32 +432,18 @@ public void onImageActivityResult(Request req) {
     }
 
     public void onVideoActivityResult(Request req, Intent intent) {
-        Uri data = null;
-
-        if (intent != null){
-            // Get the uri of the video clip
-            data = intent.getData();
-        }
-
-        if( data == null){
-            File movie = new File(getTempDirectoryPath(), "Capture.avi");
-            data = Uri.fromFile(movie);
-        }
-
-        // create a file object from the uri
-        if(data == null) {
+        if(this.videoAbsolutePath != null) {
```

Comment:
I suggest to remove the check on `videoAbsolutePath != null` ; no such check for audio or image, and the absolutePath is always defined when this method is called